moskito-webui/src/main/java/net/anotheria/moskito/webui/auth/api/AuthApiImpl.java

```diff
@@ -0,0 +1,98 @@
+package net.anotheria.moskito.webui.auth.api;
+
+import net.anotheria.anoplass.api.APIException;
+import net.anotheria.anoplass.api.APIInitException;
+import net.anotheria.moskito.webui.util.AuthCredentialsConfig;
+import net.anotheria.moskito.webui.util.CryptUtil;
+import net.anotheria.moskito.webui.util.WebUIConfig;
+import org.bouncycastle.crypto.InvalidCipherTextException;
+import org.bouncycastle.crypto.params.KeyParameter;
+
+import java.util.Arrays;
+
+/**
+ * Implementation of Auth API
+ */
+public class AuthApiImpl implements AuthApi {
+
+    /**
+     * Returns user authorization credentials
+     * @return array of user credentials
+     */
+    private AuthCredentialsConfig[] getAuthCredentials(){
+        return WebUIConfig.getInstance().getAuthCredentials();
+    }
+
+    /**
+     * Provides decryption/encryption methods.
+     */
+    private CryptUtil cryptUtil;
+
+    @Override
+    public String encryptUserCredentials(UserAO user) throws APIException {
+        try {
+            return cryptUtil.encrypt(user.getUsername() + "#" + user.getPassword());
```

Comment:
"The content of the cookie is user's password and username plus some random value. "
Where is the random value?